src/autopas/selectors/tuningStrategy/ruleBasedTuning/RuleLanguage.g4

```diff
@@ -0,0 +1,143 @@
+// This file describes the grammar of the rule language used by rule based tuning. The parsing code is generated using
+// the ANTLR plugin for CLion. It is generated into parser_generated/ and committed to the repository. Changing this
+// file requires manual regeneration of the code into this directory, and also committing (and formatting) the generated
+// files.
+
+grammar RuleLanguage;
+
+program
+                : statement+ ;
+
+COMMENT         : '#' .*? '\r'? '\n' -> skip
+                ;
+
+WS
+                : [ \r\n\t]+ -> skip
+                ;
+
+Container_opt
+                : 'VerletClusterLists'
+                | 'VerletListsCells'
+                | 'LinkedCells'
+                | 'DirectSum'
+                | 'LinkedCellsReferences'
+                | 'VarVerletListsAsBuild'
+                | 'VerletLists'
+                | 'PairwiseVerletLists'
+                ;
+
+Traversal_opt
+                : 'ds_sequential' | 'lc_c01' | 'lc_c01_combined_SoA' | 'lc_c01_cuda' | 'lc_c04' | 'lc_c04_HCP'
+                | 'lc_c04_combined_SoA' | 'lc_c08' | 'lc_c18' | 'lc_sliced' | 'lc_sliced_balanced' | 'lc_sliced_c02'
+                | 'vcc_cluster_iteration_cuda' | 'vcl_c01_balanced' | 'vcl_c06' | 'vcl_cluster_iteration' | 'vcl_sliced'
+                | 'vcl_sliced_balanced' | 'vcl_sliced_c02' | 'vl_list_iteration' | 'vlc_c01' | 'vlc_c18' | 'vlc_sliced'
+                | 'vlc_sliced_balanced' | 'vlc_sliced_c02' | 'vlp_c01' | 'vlp_c18' | 'vlp_sliced'
+                | 'vlp_sliced_balanced' | 'vlp_sliced_c02' | 'vvl_as_built'
+                ;
+
+Load_estimator_opt
+                : 'none'
+                | 'neighbor-list-length'
+                | 'squared-particles-per-cell'
+                ;
+
+Data_layout_opt
+                : 'AoS'
+                | 'SoA'
+                ;
+
+Newton3_opt
+                : 'enabled'
+                | 'disabled'
+                ;
+
+Bool_val
+                : 'false'
+                | 'true'
+                ;
+
+Configuration_property
+                : 'container'
+                | 'traversal'
+                | 'dataLayout'
+                | 'newton3'
+                | 'loadEstimator'
+                | 'cellSizeFactor'
+                ;
+
+DIGIT: '0'..'9';
+
+Unsigned_val
+                : DIGIT+
+                ;
+
+Double_val
+                : DIGIT+ '.' DIGIT*
+                ;
+
+unsigned_val
+                : Unsigned_val
+                | DIGIT
+                ;
+
+literal
+                : '"' Traversal_opt '"'
```

Comment:
I think the grammar is ambiguous without them, because a literal could then also be a variable, and both are valid expressions. Also, this probably makes future changes to the grammar easier. For example, in the future, we might want to not include all container, data layout, etc. options in this file, use arbitrary strings in quotes here instead, and resolve to the actual option after parsing. Without quotes, there is no way to distinguish these from variables, then.

Reply:
Sounds reasonable. Feels like we should note this somewhere. At least your project report ;)